client/tokenized-payment-request/transformers/wc-to-stripe.js

```diff
@@ -13,26 +13,37 @@ import { __ } from '@wordpress/i18n';
 export const transformCartDataForDisplayItems = ( cartData ) => {
 	const displayItems = cartData.items.map( ( item ) => ( {
 		amount: parseInt( item.prices.price, 10 ),
-		// TODO: should we also add variation attributes?
-		label: [ item.name, item.quantity > 1 && ` (x${ item.quantity })` ]
+		label: [
+			item.name,
+			item.quantity > 1 && `(x${ item.quantity })`,
+			item.variation &&
+				item.variation
+					.map(
+						( variation ) =>
+							`${ variation.attribute }: ${ variation.value }`
+					)
+					.join( ', ' ),
+		]
 			.filter( Boolean )
-			.join( '' ),
-		pending: true,
+			.join( ' ' ),
 	} ) );
 
-	if ( cartData.totals.total_tax ) {
+	const taxAmount = parseInt( cartData.totals.total_tax || '0', 10 );
+	if ( taxAmount ) {
 		displayItems.push( {
-			amount: parseInt( cartData.totals.total_tax, 10 ),
```

Comment:
Moved the `parseInt` outside of the `if` condition, in case the string is `'0'`.